lib/rspec/support/method_signature_verifier.rb

```diff
@@ -26,9 +26,13 @@ def non_kw_args_arity_description
         end
       end
 
-      def valid_non_kw_args?(positional_arg_count)
+      def valid_non_kw_args?(positional_arg_count, optional_max_arg_count=nil)
```

Comment:
You should be able to set the default for `optional_max_arg_count` to `positional_arg_count` which should eliminate the `||=` below.